ReactiveTask/Task.swift

```diff
@@ -261,32 +385,76 @@ public func launchTask(taskDescription: TaskDescription, standardOutput: SinkOf<
+					|> flatMap(.Merge) { stderrPipe -> SignalProducer<TaskEvent<NSData>, ReactiveTaskError> in
+						let stderrProducer = aggregateDataReadFromPipe(stderrPipe)
+
+						return SignalProducer { observer, disposable in
+							let (stdoutAggregated, stdoutAggregatedSink) = SignalProducer<NSData, ReactiveTaskError>.buffer(1)
+							let (stderrAggregated, stderrAggregatedSink) = SignalProducer<NSData, ReactiveTaskError>.buffer(1)
+
+							stdoutProducer.startWithSignal { signal, signalDisposable in
+								disposable += signalDisposable
+
+								signal.observe(next: { readData in
+									switch readData {
+									case let .Chunk(data):
+										sendNext(observer, .StandardOutput(data))
+
+									case let .Aggregated(data):
+										sendNext(stdoutAggregatedSink, data)
+									}
+								}, error: { error in
+									sendError(observer, error)
+									sendError(stdoutAggregatedSink, error)
+								}, completed: {
+									sendCompleted(stdoutAggregatedSink)
+								}, interrupted: {
+									sendInterrupted(stdoutAggregatedSink)
+								})
+							}
 
-						let terminationStatusProducer = SignalProducer<Int32, NoError> { observer, disposable in
-							task.terminationHandler = { task in
-								sendNext(observer, task.terminationStatus)
-								sendCompleted(observer)
+							stderrProducer.startWithSignal { signal, signalDisposable in
+								disposable += signalDisposable
+
+								signal.observe(next: { readData in
+									switch readData {
+									case let .Chunk(data):
+										sendNext(observer, .StandardError(data))
+
+									case let .Aggregated(data):
+										sendNext(stderrAggregatedSink, data)
+									}
+								}, error: { error in
+									sendError(observer, error)
+									sendError(stderrAggregatedSink, error)
+								}, completed: {
+									sendCompleted(stderrAggregatedSink)
+								}, interrupted: {
+									sendInterrupted(stderrAggregatedSink)
+								})
 							}
 
 							task.standardOutput = stdoutPipe.writeHandle
 							task.standardError = stderrPipe.writeHandle
 
-							if disposable.disposed {
-								stdoutPipe.closePipe()
-								stderrPipe.closePipe()
-
-								// Clean up the stdin pipe in a roundabout way.
-								stdinProducer.startWithSignal { signal, signalDisposable in
-									signalDisposable.dispose()
+							task.terminationHandler = { task in
+								let terminationStatus = task.terminationStatus
+								if terminationStatus == EXIT_SUCCESS {
+									disposable += stdoutAggregated
+										|> map { data in .Success(Box(data)) }
+										|> start(observer)
+								} else {
+									disposable += stderrAggregated
+										|> flatMap(.Concat) { data in
+											let errorString = (data.length > 0 ? String(UTF8String: UnsafePointer<CChar>(data.bytes)) : nil)
+											return SignalProducer(error: .ShellTaskFailed(exitCode: terminationStatus, standardError: errorString))
+										}
+										|> start(observer)
```

Comment:
When `terminationStatus != EXIT_SUCCESS`, signal from `stderrProducer` does not wait signal from `stdoutProducer`. Signal from other hand does not be needed, but should be waited.
I guess this cause segmentation fault of https://github.com/Carthage/Carthage/issues/497

changed comment position